### PR TITLE
Transducer v4.1 (update #2444)

### PR DESCRIPTION
**Pull request by b-flo**
Hi,

This PR fixes/completes a bunch of things related to the last update for transducer in ESPnet1. Sorry it took so long, november was quite complicated.

- Add support for relative positional encoding in input layers (VGG2L, linear) and TDNN
- Fix input layer selection in custom model building
- Fix a bug in initializer method when RNN (enc) and Transformer (dec) are used.
- Fix some docs and variable names etc.
- Extend test coverage
- VIVOS: update configs + results for transformer-based and conformer-based transducer models
- VIVOS: add config + results for conformer MTL
- Voxforge: add config + updated results for conformer-rnn transducer
- Commonvoice (Czech): add config + results for conformer-rnn transducer
- Commonvoice (Welsh): add config + results for conformer-rnn transducer

P.S: It will be the last time I update transducer-related works in ESPnet1 (outside general maintenance), I'll swap completely to ESPnet2.

### CODE REVIEW

**Comment by sw005320** [comment]
@pengchengguo, could you review this PR?

**Review by pengchengguo — COMMENTED** [review]
Should we remove the `pos_enc_class` parameter in the definition of function `build_conformer_block`? It seems like we don't use it in the following codes.

```
def build_conformer_block(
    block_arch,
    self_attn_class,
    pos_enc_class,
    pw_layer_type,
    pw_activation_type,
    conv_mod_activation_type,
):
```

File: espnet/nets/pytorch_backend/transducer/tdnn.py [hunk 1]
```diff
@@ -49,40 +49,81 @@ def __init__(
...
+        if isinstance(x_input, tuple):
+            xs, pos_emb = x_input[0], x_input[1]
+        else:
+            xs, pos_emb = x_input, None
+
+        xs = xs.transpose(1, 2)
         xs = self.tdnn(xs)
 
+        if self.relu:
+            xs = self.rel(xs)
```
Comment (pengchengguo) [inline espnet/nets/pytorch_backend/transducer/tdnn.py:86]: Maybe use `self.relu_fuc` or just use `F.relu`  like previous code are more easy to understand? I am not sure about it.

**Comment by b-flo** [comment]
> Should we remove the `pos_enc_class` parameter in the definition of function `build_conformer_block`? It seems like we don't use it in the following codes.
> 
> ```
> def build_conformer_block(
>     block_arch,
>     self_attn_class,
>     pos_enc_class,
>     pw_layer_type,
>     pw_activation_type,
>     conv_mod_activation_type,
> ):
> ```

You're right, I forgot to remove it, thanks!

**Comment by sw005320** [comment]
I'll also quickly review this.
It should be merged soon since this is a minor change.
Sorry about it, @b-flo...

**Review by sw005320 — COMMENTED** [review]
Many thanks.
I'm familiar with the voxforge Italian numbers and the results are very encouraging.
Great job, @b-flo!
I have a couple of minor comments.
- what do `ct` and `tt` mean?
- Do you have some comments about the batch norm in the TDNN block? Is it stably working?

**Comment by b-flo** [comment]
> what do ct and tt mean?

ct stands for conformer-transducer and tt for transformer-transducer, sorry I should have changed that.

> Do you have some comments about the batch norm in the TDNN block? Is it stably working?

Batch-norm seems stable from what I experimented. I want to do more experiments with TDNNs but current implementation is too slow and use to much memory. I'll try TDNN-F when I have time or implement it at c++ level.

**Comment by sw005320** [comment]
OK, thanks!
Year, since the other config files do not use the abbreviations, you can change them like that.

**Review by Peidong-Wang — COMMENTED** [review]
LGTM in general.

**Comment by sw005320** [comment]
> > what do ct and tt mean?
> 
> ct stands for conformer-transducer and tt for transformer-transducer, sorry I should have changed that.
> 

I want to ask you to do it in the follow-up PR.
I'll merge it now.
Many thanks, @b-flo!
